go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -784,3 +784,30 @@ func (bnc *BaseNetworkController) isLocalZoneNode(node *kapi.Node) bool {
 func (bnc *BaseNetworkController) isLayer2Interconnect() bool {
 	return config.OVNKubernetesFeature.EnableInterconnect && bnc.NetInfo.TopologyType() == types.Layer2Topology
 }
+
+func (bnc *BaseNetworkController) nodeZoneClusterChanged(oldNode, newNode *kapi.Node, newNodeIsLocalZone bool) bool {
+	// Check if the annotations have changed. Use network topology and local params to skip unecessary checks
+
+	// NodeIDAnnotationChanged and NodeTransitSwitchPortAddrAnnotationChanged affects local and remote nodes
+	if util.NodeIDAnnotationChanged(oldNode, newNode) {
+		return true
+	}
+
+	if util.NodeTransitSwitchPortAddrAnnotationChanged(oldNode, newNode) {
+		return true
+	}
+
+	// NodeGatewayRouterLRPAddrAnnotationChanged would not affect layer3 secondary network
+	if !bnc.IsSecondary() && util.NodeGatewayRouterLRPAddrAnnotationChanged(oldNode, newNode) {
+		return true
+	}
+
+	if newNodeIsLocalZone {
```

Comment:
We could really check any node, regardless for the zone. 
I guess checking a local zone node is a convention more than anything. All nodes should eventually end up annotated with the same ID.
Such a change would probably disrupt connectivity.
And we only use the network ID on startup so syncing IC here wouldn't really reconfigure the network Id anyway.
So I guess we can just leave this check out & not support network ID changes for the time being

I am sorry if I am going in circles with this :/